docs/design/storage-node-automatic-updater.md

```diff
@@ -0,0 +1,95 @@
+# Storage Node Automatic Updater
+
+## Abstract
+
+Automatic Updater is a process separate from the Storage Node that automatically downloads the latest Storage Node binary to replace the current Storage Node binary.
+
+## Background
+
+As more Storage Node Operators join the network we must ensure these nodes have a mechanism to automatically keep their nodes software up to date.
+If a Storage Node Operator fails to keep their node up to date with the minimum version required by satellites they will no longer be selected for upload or download requests.s
+Currently we are using Docker for automatic updates but we are migrating away from docker so we need to write out own automatic update utility for Storage Nodes.
+
+## Design
+
+* Contacts Version Server to determine if Storage Node needs to be updated.
+* Downloads latest Storage Node binary.
+* Validate binary signature.
+* Stop the Windows Service for Storage Node
+* Replaces current Storage Node binary with latest Storage Node binary.
+* Kills current Storage Node binary process.
```

Comment:
This bullet should be removed. The process should be already stopped at this moment.

Reply:
No, the order should as is. First, you swap the binaries and then restart the process.

Reply:
You may not be able to do this on Windows. Very often (if not every time) I cannot overwrite the binary because it is locked by the running process.

Also, if we download and extract the binary to a temp directory, then swapping the binary from the temp directory would be a quick operation and the delay will be insignificant.

Reply:
You rename `current.exe -> old.exe` then `new.exe -> current.exe`. This usually works. Yes, you cannot delete files in use, however, you can rename them.

Reply:
Oh, I didn't know about this trick. OK, I agree with you.

Reply:
But, I guess, since we are using a separate binary, then this wouldn't be problematic. The renaming approach might have been due to needing to update the same binary, where the updater is running.

Reply:
However, can we avoid the word "kill" here? This should be a graceful shutdown.

Reply:
stop and start is usually used for services

Reply:
Updated the writing significantly in multiple parts.